### ISSUE TITLE
Card list issue and how-to switch between lists is missing

### ISSUE THREAD

**Issue by russellfei** [org-trello/org-trello#297]
Environment:
- emacs 24.5.7
- org-trello from melpa 0.7.5

Issue:
- Cannot pull previous trello board, why missing? It says like "#TODO" problem, but **I've never found anything like "`C-c C-c` to refresh"!
- Only can create board from new local file, and **NO assigment of card list**, the only available TAG is the color label :-\
- two versions of API, `org-trello-xxx` and `org-trello/xxx`
- `help-describe-key-binding` is not complete

Advice:
- why not rename like **"pull"** and **"push"** just like git?
- Please **ADD** card list configuration setting in documentation
- It's happy to see org-tello with more clearly usage directions

Really looking forward but very upset now....

**Comment by ardumont** [comment]
Hello,

A note, I'd prefer you issue dedicated ones instead of a bunch in one go.
It's far too complicated to answer because each can have a great timeline.

> how to switch between list.

I suppose list means trello list in the trello board (I see it as org keywords).

To switch a card to a board, the org-keyword needs to be changed (using org's traditional way - C-c t or C-c C-t).
And when you sync to trello, the card will be moved to the corresponding list.

> Cannot pull previous trello board, why missing? 

What does `pull previous board` mean?
The previous version?

> It says like "#TODO" problem, but **I've never found anything like "C-c C-c to refresh"!

Can you please describe the use case?

> Only can create board from new local file, and NO assigment of card list, the only available TAG is the color label :-\
> two versions of API, org-trello-xxx and org-trello/xxx

Nope `org-trello/` is deprecated.
It's aliases i kept for not upsetting previous users...
When I adapted org-trello to respect emacs conventions #238 

> help-describe-key-binding is not complete

Well, nobody's perfect.
What's missing?

This one is simple to do, here is the code - https://github.com/org-trello/org-trello/blob/master/org-trello-setup.el#L185

>     why not rename like "pull" and "push" just like git?

There is an issue open on this #284 

>     Please ADD card list configuration setting in documentation

Can you elaborate?
(Also, there is a repository for this org-trello/org-trello.github.io)

...

Oh, you mean, how to add actual trello list to the board?
You cannot right now.

The use case did not appear yet.
But indeed, this seems pretty useful.

workaround:
You can update the trello board with a new list using the browser.
And then update the board inside the org buffer.
M-x org-trello-update-board-metadata
This will update the buffer with the board's current metadata.

>     It's happy to see org-tello with more clearly usage directions

I did videos but i don't have time to update them.
Still the dsl changed but the usage did not.

> Really looking forward but very upset now....

All things considered, I was upset when org-trello did not exist.
Back then, i needed to use trello (which is hype and cool and all, i'm just not a mouse guy).

Then I started it and now it's better but as always nothing is perfect.

Still, as I have less time to maintain it, people contributions are the way to go.
And issues are great but hints to the implementation is great too...

To sum up, with people's help, it could go a long way.

Cheers,

**Comment by ardumont** [comment]
Also, `C-c o B - M-x org-trello-bug-report` could help sum up the environment part.

(and this one is referenced in `C-c o h - M-x org-trello-help-describing-bindings` :D)

**Comment by russellfei** [comment]
@ardumont Let me help you improve this, I'll explain those not clearly explained above.
To be frank, I was quite upset after carefully read your document because there's no interaction between trello web and local org file.

Always, I think org-trello is the superstar of mobile app integration with GTD and collaboration, it shouldn't be such imperfect:
- I can only create and upload new board, I cannot synchronize boards that is created by web trello to a local org file. So I think the pull and push - the name used in git, `pull` might be more comprehensive to install local org-mode file from trello, and `push` means to create/sync local  org-mode file to trello.
- The most important feature that create new board with a bunch of lists according to `#+TODO: TODO | DONE CANCELLED` needs to be set before you `M-x`; otherwise, it will automatically create many lists, and you CANNOT delete lists! (I'm really frustrated with this, you know, I have to close the board...) According to my trials, if you want to customize lists in initialization, `#+TODO: TODO | DONE CANCELLED` should be claimed before initialization, while this is missing in documents. Althougth FAQ gives some message about this, but `C-c C-c` to refresh the list setting doesn't work.



Regards,
Rus

**Comment by ardumont** [comment]
Hello,

> @ardumont Let me help you improve this, I'll explain those not clearly explained above.

Thanks.

> Always, I think org-trello is the superstar of 

mmph, I never had so much ambition but thanks for the vote of confidence.

> mobile app

just curious, why mobile?

> integration with GTD and collaboration, it shouldn't be such imperfect:

ah yes, imperfections... This is but the land of opportunities to create, improve or contribute to something greater than one's self!

And thanks for the issues report, this helps!

 >    I can only create and upload new board, I cannot synchronize boards that is created by web trello to a local org file. So I think the pull and push - the name used in git, it might be more comprehensive that pull means to install local org-mode file from trello, and push means to create/sync local org-mode file to trello.

Yes, some help on getting the synchronization right would be awesome.
(arbox/org-sync must be a good start to check how it can be done)...

There is an issue for that #59 opened for quite some time now...
Never got the time to go and do it though...

 >   The most important feature that create new board with a bunch of lists according to #+TODO: TODO | DONE CANCELLED needs to be set before you M-x; otherwise, it will automatically create many listss, 

Or global setup in org-mode first.

Anyway, putting things in perspective here:
1. initially you did have to create them yourself and edit the org-trello buffer yourself... quite tedious and error prone, now it seems a little simpler (but not perfect)
2. org-trello is a per-need crafted software so i need to be aware of such needs to develop something (or merge a pr).

So, now i'm aware of such need.
Can you please open a dedicated issue detailing what's the detailed use case?

> and you CANNOT delete lists! (I'm really frustrated with this, you know, I have to close the board...)

I have [some code to close a list but it's not exposed](https://github.com/org-trello/org-trello/blob/master/org-trello-api.el#L145-L147) -> [org-trello does close the default lists opened by trello on a new board](https://github.com/org-trello/org-trello/blob/3718ed704094e5e5a491749f1f722d76ba4b7d73/org-trello-controller.el#L959-L972).

Use Example, in a repl (<kbd>M-x ielm RET</kdb>):
```lisp
(let ((org-trello--user-logged-in "<user>"))
  (orgtrello-controller-load-keys)
  (orgtrello-query-http-trello (orgtrello-api-close-list  "<list-id>") 'sync))
```
Note: 
- "user" is your user account
- "list-id" is the list's id that you can found in the global buffer's properties at the beginning of the buffer in the :PROPERTIES: entry

(Note for self: This also puts again the #288 since this can be contribution plugin or snippets or whatnot...)

> According to my trials, if you want to customize lists in initialization, #+TODO: TODO | DONE CANCELLED should be claimed before initialization, while this is missing in documents.

Yes, it's related to how org-mode works... or/and how you set your org-mode's definitions...
That's why all is not detailed.

This is kind of difficult to choose where to start the documentation and where to stop it without too much maintenance hassle and/or redundancy with the main mode... (or emacs for that matters...)

Then again, doc is also a repository so once you found something is missing a pr might be nice for others (including me :).

>  Althougth FAQ gives some message about this, but C-c C-c to refresh the list setting doesn't work.

I think the faq page you mention is http://org-trello.github.io/usage.html#possible-errors

Ok, this error might not be right then... I need to refresh on this.

>    C-c C-t might not be effective, I've tested this morning, you can only choose from different labels.

The labels are set up per user.
And it might be a good idea to sync them if possible with your org-trello's needs.
I don't measure well how intrusive this is.

Here's my setup https://github.com/ardumont/orgmode-pack/blob/master/orgmode-pack.el#L44-L45.

Cheers,
